### ISSUE TITLE
`lerna updated` continues to show changes after `lerna publish`

### ISSUE THREAD

**Issue by artola** [lerna/lerna#1548]
I have a rare situation:  `lerna updated` continues to show changes after `lerna publish`
If I have 5 packages, all of them are shown under `updated` even if it is not the case.
Then, I run `lerna publish` and bump the version for the 5 packages.
Then, I run `lerna updated` and it is still showing the 5 packages recently published. Git tags are in place holding the bumped versions.

Lerna 2.11 - npm 6.1.0 - Node v10.6.0 - Mac OS

Independent Mode

Originally the monorepo and Lerna worked fine for a bunch of releases.

Do you have any hint? May be some other open issue ongoing?

**Comment by evocateur** [comment]
Are the git tags annotated?

**Comment by artola** [comment]
@evocateur Yes, they are annotated as created with Lerna. In any case, I used the following command to "ensure" that they are annotated: `for tag in $(git tag -l); do git tag -a -f $tag $tag^0 -m $tag; done`
Nevertheless, `lerna updated` still shows all the packages as changed even when there is no commit after the last `lerna publish`.

¯\_(ツ)_/¯

**Comment by evocateur** [comment]
Without being able to see the actual git history, I'm afraid I can't help you any further.

The pattern you describe seems to be most common after a squash merge (or a regular merge with conflicts) mangles the lerna tags by changing the ref they originally pointed to.

**Comment by artola** [comment]
@evocateur Thanks a lot for your prompt reply.

For example: I have a package `vscode-snippets-proactive` (between others).
I have removed all the tags for this package and added just the last one:

```sh
git tag -a vscode-snippets-proactive@1.0.0-beta.6 -m "vscode-snippets-proactive@1.0.0-beta.6"
```

I can see:

`HEAD -> master, tag: vscode-snippets-proactive@1.0.0-beta.6`

```diff
diff --git a/proactive/vscode-snippets/package.json b/proactive/vscode-snippets/package.json
index 5d1c477..c833789 100644
--- a/proactive/vscode-snippets/package.json
+++ b/proactive/vscode-snippets/package.json
@@ -1,6 +1,6 @@
 {
   "name": "vscode-snippets-proactive",
-  "version": "1.0.0-beta.5",
+  "version": "1.0.0-beta.6",
   "description": "VS Code Shareable Snippets",
```

Still if I run `lerna updated` this package is shown again.

**Comment by evocateur** [comment]
I think I figured this out. TL;DR: the change detection isn't aware that the current commit has a release tag.

**Comment by jzempel** [comment]
This is a breaking change for us as `--since` is no longer reliable for identifying updates. Error here: https://travis-ci.org/zendeskgarden/react-components/builds/419480389#L2470. Here's the lerna command CI was attempting to execute: https://github.com/zendeskgarden/react-components/blob/master/utils/travis/publish-packages.js#L22

In 3.1.x is there a recommended approach for tagging locally and allowing CI to handle publish to npm?

**Comment by evocateur** [comment]
```sh
# local
npx lerna version

# remote
npx lerna publish from-git
```
